From 3b20eb8505d7eba958b07ebdccab29881c7016b5 Mon Sep 17 00:00:00 2001
From: Eric Williams
Date: Tue, 9 Feb 2016 13:46:03 -0500
Subject: Bug 486368: [GTK3] Preferences dialog background color garbled

Restore call to gtk_style_context_set_background() in order to prevent
dialog background color issues on earlier versions of GTK.

Tested on GTK3.18, 3.16, 3.14, and 2.24. AllNonBrowser JUnit tests pass
on GTK2 and GTK3.

Change-Id: I1885f7e0a3d9312c4f7fe033c9a80abf31f20182
Signed-off-by: Eric Williams <ericwill@redhat.com>
---
 bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os_custom.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os_custom.c b/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os_custom.c
index 8f5d022..032ce43 100644
--- a/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os_custom.c	
+++ b/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os_custom.c	
@@ -512,7 +512,9 @@ static void swt_fixed_realize (GtkWidget *widget) {
 	window = gdk_window_new (gtk_widget_get_parent_window (widget), &attributes, attributes_mask);
 	gtk_widget_set_window (widget, window);
 	gdk_window_set_user_data (window, widget);
-	gtk_style_context_set_background (gtk_widget_get_style_context (widget), window);
+	if (NULL != gtk_check_version (3, 18, 0)) {
+		gtk_style_context_set_background (gtk_widget_get_style_context (widget), window);
+	}
 }
 
 static void swt_fixed_map (GtkWidget *widget) {
-- 
cgit v0.11.2-4-g4a35


